src/main/kotlin/org/jitsi/jibri/util/StatusPublisher.kt

```diff
@@ -42,8 +49,10 @@ open class StatusPublisher<T> {
      * in the context of the thread which calls [publishStatus].
      */
     protected fun publishStatus(status: T) {
-        handlers.forEach { handler ->
-            handler(status)
-        }
+//        handlers.forEach { handler ->
+//            handler(status)
+//        }
+        // This will run all handlers, but only keep the ones that return true
+        handlers.retainAll { it(status) }
```

Comment:
Can you give me an example of when you want to clear a status handler?

Reply:
we use this in the helper to enable doing:
```
whenever(component).transitionsTo(state)
```
once the component has transitioned to that state, the helper no longer needs to receive status updates